src/testing/test/testing-utils.spec.ts

```diff
@@ -53,4 +54,48 @@ describe('testing-utils', () => {
       expect(() => expectFiles(fs, [MOCK_FILE_PATH, anotherFilePath])).toThrow(expectedErrorMessage);
     });
   });
+
+  describe('doNotExpectFiles', () => {
+    let fs: d.InMemoryFileSystem;
+
+    beforeEach(() => {
+      const sys = createTestingSystem();
+      fs = createInMemoryFs(sys);
+    });
+
+    it('does not throw when no file paths are provided', async () => {
+      await fs.writeFile(MOCK_FILE_PATH, MOCK_FILE_CONTENTS);
+
+      expect(() => doNotExpectFiles(fs, [])).not.toThrow();
+    });
+
+    it('does not throw when a provided file path is not found on the file system', () => {
+      expect(() => doNotExpectFiles(fs, [MOCK_FILE_PATH])).not.toThrow();
+    });
+
+    it('does not throw when the provided file paths are not found on the file system', () => {
+      expect(() => doNotExpectFiles(fs, [MOCK_FILE_PATH, 'mock/file/path/to/nowhere.ts'])).not.toThrow();
+    });
+
+    it('throws an error when a file is found on the file system', async () => {
+      await fs.writeFile(MOCK_FILE_PATH, MOCK_FILE_CONTENTS);
+
+      const expectedErrorMessage = `The following files were expected to not exist, but do:
+-${MOCK_FILE_PATH}`;
+
+      expect(() => doNotExpectFiles(fs, [MOCK_FILE_PATH])).toThrow(expectedErrorMessage);
+    });
+
+    it('throws an error when multiple files are found on the file system', async () => {
+      await fs.writeFile(MOCK_FILE_PATH, MOCK_FILE_CONTENTS);
+      const anotherFilePath = path.join('another', 'mock', 'file', 'path', 'to', 'some-file.ts');
+      await fs.writeFile(anotherFilePath, MOCK_FILE_CONTENTS);
+
+      const expectedErrorMessage = `The following files were expected to not exist, but do:
+-${MOCK_FILE_PATH}
+-${anotherFilePath}`;
+
+      expect(() => doNotExpectFiles(fs, [MOCK_FILE_PATH, anotherFilePath])).toThrow(expectedErrorMessage);
+    });
```

Comment:
would it be worthwhile to have a test that just sanity-checks that if we write two files and `doNotExpectFiles` for only one of them we get an error with only that file in it?

Reply:
👍  Done in https://github.com/ionic-team/stencil/pull/3435/commits/e4c5d176b99742272b831aea5f165badf8f141f1